sql/core/src/test/resources/sql-tests/inputs/collations.sql

```diff
@@ -90,3 +90,457 @@ select 'a' collate en_ci_ai = 'Å';
 select 'Kypper' collate sv < 'Köpfe';
 select 'Kypper' collate de > 'Köpfe';
 select 'I' collate tr_ci = 'ı';
+
+-- ConcatWs
+SELECT concat_ws(collate(' ', 'UTF8_BINARY'), collate('Spark', 'UTF8_BINARY'), collate('SQL', 'UTF8_BINARY'));
+SELECT concat_ws(collate(' ', 'UTF8_BINARY'), 'Spark', 'SQL');
+SELECT concat_ws(' ', collate('Spark', 'UTF8_BINARY'), collate('SQL', 'UTF8_BINARY'));
+SELECT concat_ws(collate(' ', 'UTF8_LCASE'), collate('Spark', 'UTF8_LCASE'), collate('SQL', 'UTF8_LCASE'));
+SELECT concat_ws(collate(' ', 'UTF8_LCASE'), 'Spark', 'SQL');
+SELECT concat_ws(' ', collate('Spark', 'UTF8_LCASE'), collate('SQL', 'UTF8_LCASE'));
+SELECT concat_ws(collate(' ', 'UNICODE'), collate('Spark', 'UNICODE'), collate('SQL', 'UNICODE'));
+SELECT concat_ws(collate(' ', 'UNICODE'), 'Spark', 'SQL');
+SELECT concat_ws(' ', collate('Spark', 'UNICODE'), collate('SQL', 'UNICODE'));
+SELECT concat_ws(collate(' ', 'UNICODE_CI'), collate('Spark', 'UNICODE_CI'), collate('SQL', 'UNICODE_CI'));
+SELECT concat_ws(collate(' ', 'UNICODE_CI'), 'Spark', 'SQL');
+SELECT concat_ws(' ', collate('Spark', 'UNICODE_CI'), collate('SQL', 'UNICODE_CI'));
+SELECT concat_ws(' ', collate('Spark', 'UTF8_LCASE'), collate('SQL', 'UNICODE'));
+
```

Comment:
also, we should have proper tests for collation type coercion & collation mismatches (for all functions)

please refer to: https://github.com/apache/spark/pull/47620#issuecomment-2278017099